swap/src/protocol/alice/event_loop.rs

```diff
@@ -258,11 +258,12 @@ where
                                 continue;
                             }
 
+                            // TODO: This seems to internally dial the peer, Alice should not dial Bob - is this wanted behaviour?
```

Comment:
I guess there could be the possibility of us losing the connection after we checked it here and before the request can get sent. So instead of buffering it actively, we should maybe track that we tried to send it and if we fail to send it (we can listen for that event), simply try again once we are connected :)